### PR TITLE
fixes #9075: sympy.limit yields incorrect result

### LINKED ISSUE #9075: sympy.limit yields incorrect result
Python 2.7.9; SymPy 0.7.6

Consider the function `g=(6**(n+1)+n+1)/(6**n+n)` where x is real and n is an integer

It is easy to show that the limit of g as n->oo is 6

Yet, look at the output of the following code:

```
from sympy import symbols, limit, oo, Pow
n=symbols('n',integer=True)
x=symbols('x',real=True)
g=(Pow(6,n+1)+n+1)/(Pow(6,n)+n)
limit(g,n,oo)
```
out[4]: 1

### PR DESCRIPTION
**Pull request by leosartaj**
fixed the limit,

```
>>> ((6**(n+1)+n+1)/(6**n+n)).limit(n, oo)
6
```

In general,

```
>>> ((6**(n+x)+n+1)/(6**n+n)).limit(n, oo)
6**x
```

Fixes #9075 
@raoulb @smichr

### CODE REVIEW

**Comment by leosartaj** [comment]
if somebody could have a look @asmeurer @raoulb @smichr @debugger22

File: sympy/core/power.py [hunk 1]
```diff
@@ -1302,7 +1302,14 @@ def _eval_as_leading_term(self, x):
         from sympy import exp, log
         if not self.exp.has(x):
             return self.func(self.base.as_leading_term(x), self.exp)
-        return exp(self.exp * log(self.base)).as_leading_term(x)
```
Comment (zanzibar7) [inline sympy/core/power.py]: Why not just `return self`?  The concept of a leading term for a sum series seems not to make sense when the variable occurs in both the base and the exponent.

**Comment by leosartaj** [comment]
If you could have a look @jksuom

**Comment by jksuom** [comment]
The 'heuristic' method of finding limits is intended primarily for those cases where serial expansions of various subexpressions exist (including Laurent series). Exponentials (and powers) do not have such expansions at infinity. Hence I believe that the correct strategy would be to let ``gruntz`` handle situations like this.

(The patch will cover this particular example, but I cannot convince myself that it would suffice in general.)

**Comment by leosartaj** [comment]
This type of limits are already solved by heuristics. Here is what I did.

During solving the original expression ``(6**(n+x)+n+1)/(6**n+n)``, ``n`` is replaced by ``1 / n`` and it's leading term is calculated.

Earlier
```
>>> ((6**(1/n+x)+1/n+1)/(6**(1/n)+1/n)).as_leading_term(n)
1
```
Fixed it to return 6**x instead.
@jksuom

**Comment by jksuom** [comment]
According to the docs ``as_leading_term`` is intended to return the leading (i.e. first nonzero) term of a series expansion. What is the series expansion you have in mind for ``6**(1/n+x)``?

**Comment by leosartaj** [comment]
This is what sympy returns

```
>> ex = 6**(1/n + x)
>>> ex.series(n)
exp((x + 1/n) * log(6))
>>> ex.as_leading_term(n)
exp(log(6) / n)
```

**Comment by jksuom** [comment]
Those lines mean that there is no series expansion. Hence there is no correct leading term either.

In fact, exponentials have an essential singularity at infinity.

**Pull request closed without merging**